File: app/src/main/java/com/maiya/leetcode/phone/service/TaskServiceBinder.java
package com.maiya.leetcode.phone.service;

import android.os.Binder;

import com.maiya.leetcode.phone.manager.CallListenerService;

import java.lang.ref.WeakReference;

/**
 * Service 管理
 */

public class TaskServiceBinder extends Binder {

    private WeakReference<CallListenerService> weakService;

    /**
     * Inject service instance to weak reference.
     */
    public void onBind(CallListenerService service) {
        this.weakService = new WeakReference<>(service);
    }

    public CallListenerService getService() {
        return weakService == null ? null : weakService.get();
    }
}

File: app/src/main/java/com/maiya/call/phone/service/TaskServiceBinder.kt
package com.maiya.call.phone.service

import android.os.Binder
import java.lang.ref.WeakReference

/**
 * Service 管理
 */
class TaskServiceBinder : Binder() {
    private var weakService: WeakReference<CallListenerService>? = null

    /**
     * Inject service instance to weak reference.
     */
    fun onBind(service: CallListenerService) {
        weakService = WeakReference(service)
    }

    val service: CallListenerService?
        get() = if (weakService == null) null else weakService!!.get()
}